Sources/TuistLoader/Up/UpRome.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// Up that updates outdated Rome dependencies.
+class UpRome: Up, GraphInitiatable {
+    /// The platforms Rome dependencies should be updated for.
+    let platforms: [Platform]
+
+    /// The cache prefix to use when downloading dependencies
+    let cachePrefix: String?
+
+    /// Up homebrew for installing Rome.
+    let upHomebrew: Upping
+
+    /// Rome instace to interact with the project Romefile setup.
+    let rome: Romeaging
+
+    /// Initializes the Carthage command.
+    ///
+    /// - Parameters:
+    ///   - platforms:      The platforms Carthage dependencies should be updated for.
+    ///   - cachePrefix:    The cachePrefix used when downloading Rome dependencies.
+    ///   - upHomebrew:     Up homebrew for installing Carthage.
+    ///   - rome:           Rome instace to interact with the projects Rome setup.
+    init(platforms: [Platform],
+         cachePrefix: String?,
+         upHomebrew: Upping = UpHomebrewTap(repositories: ["tmspzz/tap"]),
+         rome: Romeaging = Rome()) {
+        self.platforms = platforms
+        self.cachePrefix = cachePrefix
+        self.upHomebrew = upHomebrew
+        self.rome = rome
+        super.init(name: "Rome download")
+    }
+
+    /// Default constructor of entities that are part of the manifest.
+    ///
+    /// - Parameters:
+    ///   - dictionary: Dictionary with the object representation.
+    ///   - projectPath: Absolute path to the folder that contains the manifest. This is useful to obtain absolute paths from the relative paths provided in the manifest by the user.
+    /// - Throws: A decoding error if an expected property is missing or has an invalid value.
+    required convenience init(dictionary: JSON, projectPath _: AbsolutePath) throws {
+        var platforms: [Platform] = []
+        var cachePrefix: String?
+        if let platformStrings: [String] = try? dictionary.get("platforms") {
+            platforms = platformStrings.compactMap {
+                Platform(rawValue: $0)
+            }
+        }
+
+        if let cachePrefixString: String = dictionary.get("cachePrefix") {
+            cachePrefix = cachePrefixString
+        }
+        self.init(platforms: platforms, cachePrefix: cachePrefix)
+    }
+
+    /// Returns true when the command doesn't need to be run.
+    ///
+    /// - Parameters
+    ///   - projectPath: Path to the directory that contains the project manifest.
+    /// - Returns: True if the command doesn't need to be run.
+    /// - Throws: An error if the check fails.
+    override func isMet(projectPath: AbsolutePath) throws -> Bool {
+        if try !upHomebrew.isMet(projectPath: projectPath) { return false }
+        return true
```

Comment:
Is there a way to know if Rome has already pulled those pre-compiled dependencies? If so, we should add the logic here instead of returning true. Otherwise, every time someone does `tuist up` they'll end up downloading the frameworks even if they already exist locally.

Reply:
According to the [rome's documentation](https://github.com/tmspzz/Rome#usage), a `list` command looks promising:

```bash
rome list --help
...
Lists frameworks in the cache and reports cache misses/hits, according to the
local Cartfile.resolved. Ignores dSYMs.

--missing                List frameworks missing from the cache. Ignores dSYMs
--present                List frameworks present in the cache. Ignores dSYMs.
```

Maybe we could try to leverage one of these flags? 🤔

Reply:
Hi @mollyIV thank you for looking into this. I am just about to push a change up now to use rome list that will check for any missing frameworks in advance.